gobblin-config-management/gobblin-config-client/src/main/java/gobblin/config/client/ConfigClient.java

```diff
@@ -0,0 +1,352 @@
+   * @throws VersionDoesNotExistException: Required version does not exist anymore ( may get deleted by retention job )
+   */
+  public Collection<URI> getImports(URI configKeyUri, boolean recursive)
+      throws ConfigStoreFactoryDoesNotExistsException, ConfigStoreCreationException, VersionDoesNotExistException {
+    ConfigStoreAccessor accessor = this.getConfigStoreAccessor(configKeyUri);
+    ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(configKeyUri, accessor.configStore);
+    Collection<ConfigKeyPath> result;
+    
+    if(!recursive){
+      result = accessor.topologyInspector.getOwnImports(configKeypath);
+    }
+    else{
+      result = accessor.topologyInspector.getImportsRecursively(configKeypath);
+    }
+    
+    return ConfigClientUtils.getAbsoluteURI(result, accessor.configStore);
+  }
+
+  /**
+   * Get the URIs which imports the input URI
+   * 
+   * @param configKeyUri - The URI for the configuration key. 
+   * @param recursive    - Specify whether to get direct or recursively imported by links
+   * @return  the URIs which imports the input URI
+   * 
+   * @throws ConfigStoreFactoryDoesNotExistsException: if missing scheme name or the scheme name is invalid
+   * @throws ConfigStoreCreationException: Specified {@link ConfigStoreFactory} can not create required {@link ConfigStore}
+   * @throws VersionDoesNotExistException: Required version does not exist anymore ( may get deleted by retention job )
+   */
+  public Collection<URI> getImportedBy(URI configKeyUri, boolean recursive)
+      throws ConfigStoreFactoryDoesNotExistsException, ConfigStoreCreationException, VersionDoesNotExistException {
+    ConfigStoreAccessor accessor = this.getConfigStoreAccessor(configKeyUri);
+    ConfigKeyPath configKeypath = ConfigClientUtils.buildConfigKeyPath(configKeyUri, accessor.configStore);
+    Collection<ConfigKeyPath> result;
+    
+    if(!recursive){
+      result = accessor.topologyInspector.getImportedBy(configKeypath);
+    }
+    else{
+      result = accessor.topologyInspector.getImportedByRecursively(configKeypath);
+    }
+    
+    return ConfigClientUtils.getAbsoluteURI(result, accessor.configStore);
+  }
+  
+  private URI getMatchedFloorKeyFromCache(URI configKeyURI){
+    URI floorKey = this.configStoreAccessorMap.floorKey(configKeyURI);
+    if(floorKey==null) {
+      return null;
+    }
+    
+    // both scheme name and authority name should match
+    if(floorKey.getScheme().equals(configKeyURI.getScheme())){
+      // no authority/store root directory
+      if(floorKey.getAuthority()==null && configKeyURI.getAuthority()==null){
+        return floorKey;
+      }
+      
+      if(floorKey.getAuthority()==null || configKeyURI.getAuthority()==null){
+        return null;
+      }
+      
+      // both are absolute URI
+      if(floorKey.getAuthority().equals(configKeyURI.getAuthority())){
+        if(configKeyURI.getPath().startsWith(floorKey.getPath())) {
```

Comment:
`startsWith` is not enough. `"/data/databases".startsWith("/dat")` is `true`, but that is not a valid match. You need to check whether `floorKey` is an ancestor of `configKeyURI`.